### PR TITLE
refactor/bump:  Wrap revm return

### LINKED ISSUE #463: Wrap revm `Return` status to proper error.
### Describe the feature

Revm will refactor return type to be more friendly and when this happens errors should be properly wrapped inside reth

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by fkrause98**
## Changes:
Closes #463.

### CODE REVIEW

File: crates/primitives/Cargo.toml [hunk 1]
```diff
@@ -17,9 +17,10 @@ reth-rlp = { path = "../rlp", features = [
 reth-rlp-derive = { path = "../rlp/rlp-derive" }
 reth-codecs = { version = "0.1.0", path = "../storage/codecs" }
 
-revm-interpreter = { git = "https://github.com/bluealloy/revm", rev = "a05fb262d87c78ee52d400e6c0f4708d4c527f32", features = [
-    "serde",
-] }
+revm = { version = "3.0.0"}
```
Comment (gakonst) [inline crates/primitives/Cargo.toml:20]: use `revm-primitives` crate instead

File: crates/storage/codecs/Cargo.toml [hunk 2]
```diff
@@ -23,17 +23,18 @@ arbitrary = [
 [dependencies]
 bytes = "1.2.1"
 codecs-derive = { version = "0.1.0", path = "./derive", default-features = false }
-revm-interpreter = { git = "https://github.com/bluealloy/revm", rev = "a05fb262d87c78ee52d400e6c0f4708d4c527f32", features = [
-    "serde",
-] }
+revm-interpreter = {version = "1.0.0", features = ["serde"] }
+
+#     "serde",
+# ] }
```
Comment (gakonst) [inline crates/storage/codecs/Cargo.toml:29]: use revm_primitives instead, no need to have interpreter

File: crates/primitives/Cargo.toml (hunk 1 above)
Comment (fkrause98) [reply crates/primitives/Cargo.toml]: Done!

File: crates/storage/codecs/Cargo.toml (hunk 2 above)
Comment (fkrause98) [reply crates/storage/codecs/Cargo.toml]: Fixed :-)

**Comment by rakita** [comment]
Will push some changes if that is okay, to flush out the CI and error handling.

File: crates/executor/src/executor.rs [hunk 3]
```diff
@@ -371,46 +376,63 @@ where
...
+            if let Ok(ResultAndState { ref result, .. }) = out {
+                cumulative_gas_used += match &result {
+                    Success { gas_used, .. } => gas_used,
+                    Revert { gas_used, .. } => gas_used,
+                    Halt { gas_used, .. } => gas_used,
+                }
+            };
+            // cumulative_gas_used += gas_used;
+
+            if let Ok(ResultAndState { result: Success { logs, .. }, state }) = out {
```
Comment (rakita) [inline crates/executor/src/executor.rs:417]: Already made changes, just wanting to comment this to clear confusion, maybe it will be useful for future.
Initial discussion on revm returnables was here: https://github.com/bluealloy/revm/issues/309

tldr is: revm return type is specified as `Result<ResultAndState,EVMError>`.
Only if it is `EVMError`, transaction is invalid and whole block becomes invalid.

If it is Ok as in `ResultAndState`, transactions is valid and can be found inside block but `Receipt::is_success` needs to be set depending on `ExecutionResult`: If `Success` set true, if `Revert` or `Halt` return `false`

**Comment by fkrause98** [comment]
> Will push some changes if that is okay, to flush out the CI and error handling.

Sure, go ahead.

**Comment by fkrause98** [comment]
@rakita 
Would you rather to bring the changes from #1248 into this PR or close this PR and continue working on #1248?


Either way, some tests are failing, like this one:
```
thread 'types::blocks::GetBlockHeadersTests::proptest' panicked at 'assertion failed: `(left == right)`
  left: `48`,
 right: `32`', /Users/fran/.cargo/registry/src/github.com-1ecc6299db9ec823/revm-primitives-1.0.0/src/bits.rs:9:1
```
I'm looking into it.

**Comment by gakonst** [comment]
Let's move over to Dragan's PR if you can push to it @fkrause98

**Comment by rakita** [comment]
> @rakita Would you rather to bring the changes from #1248 into this PR or close this PR and continue working on #1248?
> 
> Either way, some tests are failing, like this one:
> 
> ```
> thread 'types::blocks::GetBlockHeadersTests::proptest' panicked at 'assertion failed: `(left == right)`
>   left: `48`,
>  right: `32`', /Users/fran/.cargo/registry/src/github.com-1ecc6299db9ec823/revm-primitives-1.0.0/src/bits.rs:9:1
> ```
> 
> I'm looking into it. Edit: Fixed, seems like the other failing ci checks are already fixed on the other PR.

I didn't have access to push to lambdaclass repo to pushto this PR, so I made my own branch, will merge the changes there and fix pending failing tests.

**Comment by gakonst** [comment]
superseded by #1248
